packages/bitfield/src/lib/BitField.ts

```diff
@@ -0,0 +1,279 @@
+	 */
+	public any(field: ValueResolvable<this>, bits: ValueResolvable<this>): boolean {
+		return (this.resolve(field) & this.resolve(bits)) !== this.zero;
+	}
+
+	/**
+	 * Checks whether or not `field` is a superset of or equal to `bits`.
+	 * @param field The bits to compare the bits from.
+	 * @param bits The bits to compare with.
+	 * @returns Whether or not `field` is a superset of or equal to `bits`, also denoted as `A ⊇ B`.
+	 */
+	public has(field: ValueResolvable<this>, bits: ValueResolvable<this>): boolean {
+		const resolved = this.resolve(bits);
+		return (this.resolve(field) & resolved) === resolved;
+	}
+
+	/**
+	 * Makes the complement of `field`, which is a field of all bits (of `U` or the union of all {@link Flags} bits)
+	 * that do not belong to `A`. It is the result of `U ∖ A`, or `difference(U, field)`.
+	 * @param field The bits to get the complement of.
+	 * @returns The complement of `field`, also denoted `Aᶜ` or `A'`.
+	 * @example
+	 * ```typescript
+	 * const bitfield = new BitField({
+	 * 	Read:   0b0001,
+	 * 	Write:  0b0010,
+	 * 	Edit:   0b0100,
+	 * 	Delete: 0b1000
+	 * });
+	 *
+	 * bitfield.complement(0b0100);
+	 * // 0b1011
+	 * ```
+	 */
+	public complement(field: ValueResolvable<this>): ValueType<this> {
+		return this.difference(this.mask, field);
+	}
+
+	/**
+	 * Makes the union of all bits.
+	 * @param fields The bits to create an union of.
+	 * @returns The result of combining all bits together, also denoted as `∅ ⋃ fields`.
+	 * @example
+	 * ```typescript
+	 * bitfield.union(0b0001, 0b0100);
+	 * // 0b0101
+	 *
+	 * bitfield.union(0b1100, 0b0001, 0b0010);
+	 * // 0b1111
+	 * ```
+	 * @seealso {@link https://en.wikipedia.org/wiki/Union_(set_theory)}
+	 */
+	public union(...fields: readonly ValueResolvable<this>[]): ValueType<this> {
+		let field = this.zero as ValueType<this>;
+		for (const resolvable of fields) {
+			field = (field | this.resolve(resolvable)) as ValueType<this>;
+		}
+
+		return field;
+	}
+
+	/**
+	 * Makes the intersection of all bits.
+	 * @param bitfield The first field.
+	 * @param fields The bits to intersect with `bitfield`.
+	 * @returns The result of intersecting `bitfield` with all of the `fields`, also denoted as `A ⋂ fields`.
+	 * @example
+	 * ```typescript
+	 * bitfield.intersection(0b0001, 0b0100);
+	 * // 0b0000
+	 *
+	 * bitfield.intersection(0b1100, 0b0100);
+	 * // 0b0100
+	 *
+	 * bitfield.intersection(0b1101, 0b0101, 0b1100);
+	 * // 0b0100
+	 * ```
+	 * @seealso {@link https://en.wikipedia.org/wiki/Intersection_(set_theory)}
+	 */
+	public intersection(bitfield: ValueResolvable<this>, ...fields: readonly ValueResolvable<this>[]): ValueType<this> {
+		let field = this.resolve(bitfield);
+		for (const resolvable of fields) {
+			field = (field & this.resolve(resolvable)) as ValueType<this>;
+		}
+
+		return field;
+	}
+
+	/**
+	 * Removes from `a` the bits from `b`.
+	 * @param a The first field.
+	 * @param b The bits to create an union of.
```

Comment:
This appears to be copypasted from union

Reply:
It's a leftover from an older version of the code, `b` used to be a rest parameter that called `this.union(...b)`. Needs changing, yeah.